nucliadb/nucliadb/ingest/service/writer.py

```diff
@@ -220,41 +220,50 @@ async def NewKnowledgeBox(  # type: ignore
+        First, try to get the learning configuration for the new knowledge box.
+        From there we need to extract the semantic model metadata and pass it to the create_kb method.
+        If the kb creation fails, rollback the learning configuration for the kbid that was just created.
         """
+        kbid = request.forceuuid or str(uuid.uuid4())
         release_channel = get_release_channel(request)
         request.config.release_channel = release_channel
-        kbid = await self.proc.create_kb(
-            request.slug,
-            request.config,
-            parse_model_metadata(request),
-            forceuuid=request.forceuuid,
-            release_channel=release_channel,
-        )
-        if not request.learning_config:
-            # Since we depend on the NucliaDB admin to adapt to these changes and send the learning
-            # config on KB creation, we will log a warning and continue for now. Eventually we will
-            # want to enforce that the learning config is always provided on KB creation.
-            logger.warning("No learning configuration set for KB", extra={"kbid": kbid})
-            return kbid
+        lconfig = await learning_config.get_configuration(kbid)
+        lconfig_created = False
+        if lconfig is None:
+            if request.learning_config:
+                # We parse the desired configuration from the request and set it
+                config = json.loads(request.learning_config)
+            else:
+                # We set an empty configuration so that learning chooses the default values.
+                config = {}
+                logger.warning(
+                    "No learning configuration provided. Default will be used.",
+                    extra={"kbid": kbid},
+                )
+            lconfig = await learning_config.set_configuration(kbid, config=config)
+            lconfig_created = True
         try:
-            lconfig = json.loads(request.learning_config)
-            await learning_config.set_configuration(kbid, lconfig)
-            logger.info("Learning configuration set", extra={"kbid": kbid})
+            await self.proc.create_kb(
+                request.slug,
+                request.config,
+                parse_model_metadata(lconfig),
+                forceuuid=kbid,
+                release_channel=release_channel,
+            )
+            return kbid
         except Exception:
-            # Rollback KB that was just created
+            # Rollback learning config for the kbid that was just created
             try:
-                await self.proc.delete_kb(kbid=kbid)
+                if lconfig_created:
+                    await learning_config.delete_configuration(kbid)
             except Exception:
                 logger.warning(
-                    "Could not rollback KB", exc_info=True, extra={"kbid": kbid}
+                    "Could not rollback learning configuration",
```

Comment:
Should we report to sentry or log an error? A failure here will leave orphan configuration on learning side